kaggle_environments/agent.py

```diff
@@ -112,20 +112,32 @@ def act(self, observation, timeout=10):
         if self.agent is None:
             self.agent = build_agent(self.raw, self.environment)
             # Add in the initialization timeout since this is the first time this agent is called
-            timeout += self.configuration.agentTimeout
+            timeout += self.environment.configuration.agentTimeout
 
         args = [
            structify(observation),
-           structify(self.configuration)
+           structify(self.environment.configuration)
         ][:self.agent.__code__.co_argcount]
 
-        try:
-            action = self.agent(*args)
-        except Exception as e:
-            action = e
+        with StringIO() as out_buffer, StringIO() as err_buffer, redirect_stdout(out_buffer), redirect_stderr(err_buffer):
```

Comment:
Here we capture stdout and stderr for the agent invocation and trim to a max of 1024 characters each